### PR TITLE
WA-377 adds dynamic categories instead of old enum

### PR DESCRIPTION
**Pull request by smileman**
**Description**

**Issues that should be CLOSED by merge of this PR:**
* Fixes #
https://guestlogix.atlassian.net/browse/WA-377
https://guestlogix.atlassian.net/browse/SDK-51

**Related issues that should be LINKED to this PR:**
* Connects #

### CODE REVIEW

File: traveler_ui_kit/build.gradle [hunk 1]
```diff
@@ -38,8 +38,8 @@ android {
 dependencies {
 
     //Traveler SDK
-//    api project(path: ':traveler_core_kit')
-    api "com.guestlogix:traveler_core_kit:${traveler_core_version}"
+    api project(path: ':traveler_core_kit')
+    //api "com.guestlogix:traveler_core_kit:${traveler_core_version}"
```
Comment (smileman) [inline traveler_ui_kit/build.gradle]: Asked @amirziarati how to correctly reference new traveler_core_kit version here.

Comment (amirziarati) [reply traveler_ui_kit/build.gradle]: the path dependency is only for testing. you should push it to server. whenever you changed core you push your changes and then when the changes are merged to master a new core build is ready to use. then you ping me and i push a new ui sdk with updated core version or you can make a branch change the ui core version to latest and send a pr.

Comment (smileman) [reply traveler_ui_kit/build.gradle]: Ok. 
Splitted this branch into two separate branches.

Now this PR contains only core-sdk changes. When it merged - I'll create second PR for ui-sdk

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/CategoryFacet.java [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-import com.guestlogix.travelercorekit.utilities.ObjectMappingFactory;
-
 import com.guestlogix.travelercorekit.utilities.JSONObjectGLX;
+import com.guestlogix.travelercorekit.utilities.ObjectMappingFactory;
 
 import java.io.Serializable;
 
 public class CategoryFacet implements Serializable {
-    private BookingItemCategory category;
+    private String category;
```
Comment (amirziarati) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/CategoryFacet.java]: backend is sending category as String here :/ ? why didt they change here ? its so messy

Comment (amirziarati) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/CategoryFacet.java]: i asked them they havent changed the model here. put a todo here. we should fix this when they fixed it on their side.

Comment (smileman) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/CategoryFacet.java]: added TODO

**Pull request merged**